rust/extractor/src/crate_graph.rs

```diff
@@ -178,20 +183,107 @@ fn emit_module_children(
         .collect()
 }
 
+fn emit_reexport(
+    crate_graph: &CrateGraph,
+    db: &dyn HirDatabase,
+    trap: &mut TrapFile,
+    uses: &mut HashMap<String, trap::Label<generated::Item>>,
+    import: ImportOrGlob,
+    name: &str,
+) {
+    let (use_, idx) = match import {
+        ImportOrGlob::Glob(import) => (import.use_, import.idx),
+        ImportOrGlob::Import(import) => (import.use_, import.idx),
+    };
+    let def_db = db.upcast();
+    let loc = use_.lookup(def_db);
+    let use_ = &loc.id.item_tree(def_db)[loc.id.value];
+
+    use_.use_tree.expand(|id, path, kind, alias| {
+        if id == idx {
+            let mut path_components = Vec::new();
+            match path.kind {
+                PathKind::Plain => (),
+                PathKind::Super(0) => path_components.push("self".to_owned()),
+                PathKind::Super(n) => {
+                    path_components.extend(std::iter::repeat_n("parent".to_owned(), n.into()));
+                }
+                PathKind::Crate => path_components.push("crate".to_owned()),
+                PathKind::Abs => path_components.push("".to_owned()),
+                PathKind::DollarCrate(_) => path_components.push("$crate".to_owned()),
+            }
+            path_components.extend(path.segments().iter().map(|x| x.as_str().to_owned()));
+            match kind {
+                ImportKind::Plain => (),
+                ImportKind::Glob => path_components.push(name.to_owned()),
```

Comment:
Can you elaborate on why we push `name` and not make a glob import instead?

Reply:
I think explicit imports are more clear and easier to interpret. However, if you prefer `*` imports then it is easy to change.

Reply:
Explicit imports are easier, I just wondered how a single `name` can make it up for a full glob import?

Reply:
We iterate over all types and values that are in the "item scope", for the types/values that are imported (possibly as part of a glob import) we generate a `use`  statement.

Reply:
Just see the `pub use` statements in the test case and how they end up in the output of `modules.ql`.